tests/Feature/Filament/Resources/LocationResourceTest.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Feature\Filament\Resources;
+
+use App\Domain\Locations\Models\Location;
+use App\Domain\Locations\Models\LocationImage;
+use App\Domain\Trips\Models\Trip;
+use App\Filament\Resources\LocationResource\Pages\CreateLocation;
+use Livewire\Livewire;
+use PHPUnit\Framework\Attributes\Test;
+use Tests\TestCase;
+
+class LocationResourceTest extends TestCase
+{
+    #[Test]
+    public function can_store_images_with_location(): void
+    {
+        $trip = Trip::factory()->create();
+
+        Livewire::test(CreateLocation::class)
+            ->fillForm([
+                'trip_id' => $trip->id,
+                'name' => 'Location Name',
+                'latitude' => 50,
+                'longitude' => 60,
+                'images' => [
+                    '0' => [
+                        'path' => [
+                            'location/location-image.jpg',
+                        ],
+                    ],
+                ],
+            ])
+            ->call('create')
+            ->assertHasNoFormErrors();
+
+        $this->assertDatabaseHas(Location::class, [
+            'trip_id' => $trip->id,
+            'name' => 'Location Name',
+            'latitude' => 50,
+            'longitude' => 60,
+        ]);
+
+        $this->assertDatabaseHas(LocationImage::class, [
+            'path' => 'location/location-image.jpg',
+            'name' => 'location-image.jpg',
+        ]);
```

Comment:
Could we maybe add a storage test as well here. By default at the moment this will store the default disk. Should we create a custom disk for this so it doesn't get stored in the root level, if this was to be stored on presumably something like s3?

Good to see the test btw 🙌🏼

Reply:
Also given we are using a repeater maybe the test could show multiple images being uploaded rather than just a single one?

Reply:
I've updated the test for the repeater and added the `location_images_filesystem` disk (it was already set up). Any suggestions on a way to test the storage? It looks the filament component uploads the files when attaching them to the form - not when saving.

Reply:
I've not done this myself but having a look around I'm not sure if this helps at all? https://github.com/filamentphp/filament/discussions/5861

I've found the discussions page in general on Github is quite useful for things like things. 

If not then happy to approve as all looks good 👍🏼

Reply:
Yeah no luck with this. I think the problem is because when submitting the form via the ui the image already exists in temp storage when it got uploaded via the filament file component.